### PR TITLE
Arm developer mode if needed

### PR DESCRIPTION
**Pull request by dhinakg**
TODO:
- [x] Better description for the alert popup
- [x] Make the reboot button work
- [x] Add button to turn on developer mode in settings

~~I also added whether the app requires developer mode on or not in the app info section. Feel free to remove it if you wish~~
~~- Maybe hide this on iOS 15 and below~~
Ended up removing this

### CODE REVIEW

**Comment by opa334** [comment]
There's more entitlements that require dev mode than just `get-task-allow`. Also keep fallback entitlements in mind.

**Comment by dhinakg** [comment]
> There's more entitlements that require dev mode than just `get-task-allow`.

Fixed, based on [Jonathan Levin's slides](https://newosxbook.com/bonus/0x41-16.pdf).

> Also keep fallback entitlements in mind.

The code in RootHelper already takes care of this, but I forgot about the app info section, which doesn't account for it.

Do you have a preference as to keeping the developer mode indicator that I added into the app info section? If not, I'll just remove it; it was useful for development, but on second thought, it's not very useful for the end user.

File: TrollStore/TSSettingsListController.m [hunk 1]
```diff
@@ -55,6 +55,16 @@ - (void)viewDidLoad
...
+
+	if (@available(iOS 16, *))
+	{
+		_devModeEnabled = spawnRoot(rootHelperPath(), @[@"check-dev-mode"], nil, nil) == 0;
+	}
+	else
+	{
+		_devModeEnabled = YES;
+	}
+	[self reloadSpecifiers];
```
Comment (dhinakg) [inline TrollStore/TSSettingsListController.m:67]: Is this good practice

**Comment by dhinakg** [comment]
This should be good to go now. The two ways to trigger enabling developer mode if it's off are:
- Installing an app that requires developer mode
- Pressing the button at the top of settings (still under the update version button though)

File: TrollStore/TSSettingsListController.m (hunk 1 above)
Comment (opa334) [reply TrollStore/TSSettingsListController.m:67]: probably not

**Pull request merged**